lib/ramble/ramble/pipeline.py

```diff
@@ -521,9 +512,136 @@ def _execute(self):
             executor(*exec_args)
 
 
+class PublishDeploymentPipeline(Pipeline):
+    """class for the `prepare-deployment` pipeline"""
+
+    name = 'publishdeployment'
+    index_filename = 'index.json'
+    index_namespace = 'deployment_files'
+    tar_extension = '.tar.gz'
+
+    def __init__(self, workspace, filters,
+                 create_tar=False, upload_url=None,
+                 deployment_name=None):
+        super().__init__(workspace, filters)
+        self.action_string = 'Publishing deployment of'
+        self.require_inventory = True
+        self.create_tar = create_tar
+        self.upload_url = upload_url
+        self.object_repo_name = 'object_repo'
+
+        if deployment_name:
+            workspace.deployment_name = deployment_name
+            self.deployment_name = deployment_name
+        else:
+            self.deployment_name = workspace.name
+
+    def _execute(self):
+        configs_dir = os.path.join(self.workspace.named_deployment,
+                                   ramble.workspace.workspace_config_path)
+        fs.mkdirp(configs_dir)
+
+        _copy_tree(self.workspace.config_dir, configs_dir)
+
+        aux_software_dir = os.path.join(configs_dir, ramble.workspace.auxiliary_software_dir_name)
+        fs.mkdirp(aux_software_dir)
+        aux_repo_conf = os.path.join(aux_software_dir, 'repos.yaml')
+        if os.path.exists(aux_repo_conf):
+            f = open(aux_repo_conf, 'a')
+        else:
+            f = open(aux_repo_conf, 'w+')
+            f.write('repos:\n')
+        f.write(f'- ../../{self.object_repo_name}\n')
+
+        for conf_file in ['repos', 'modifier_repos']:
+            repos_config = os.path.join(configs_dir, conf_file + '.yaml')
+            if os.path.exists(repos_config):
+                f = open(repos_config, 'a')
+            else:
+                f = open(repos_config, 'w+')
+                f.write('repos:\n')
```

Comment:
should this be `repos` or `f{conf_file}` or similar? 

I think the modifier repo needs the string `modifier_repos` here (https://github.com/GoogleCloudPlatform/ramble/blob/develop/lib/ramble/ramble/schema/modifier_repos.py#L18)

Reply:
Yeah, I think you're right. It should be a string instead of just `repos`.